src/libraries/Microsoft.Extensions.Primitives/ref/Microsoft.Extensions.Primitives.cs

```diff
@@ -54,22 +54,26 @@ public partial interface IChangeToken
         public System.ReadOnlyMemory<char> AsMemory() { throw null; }
         public System.ReadOnlySpan<char> AsSpan() { throw null; }
         public System.ReadOnlySpan<char> AsSpan(int start) { throw null; }
+        [System.Diagnostics.CodeAnalysis.MemberNotNull(nameof(Buffer))]
+        [System.Diagnostics.CodeAnalysis.MemberNotNull(nameof(Value))]
         public System.ReadOnlySpan<char> AsSpan(int start, int length) { throw null; }
         public static int Compare(Microsoft.Extensions.Primitives.StringSegment a, Microsoft.Extensions.Primitives.StringSegment b, System.StringComparison comparisonType) { throw null; }
         public bool EndsWith(string text, System.StringComparison comparisonType) { throw null; }
         public bool Equals(Microsoft.Extensions.Primitives.StringSegment other) { throw null; }
         public static bool Equals(Microsoft.Extensions.Primitives.StringSegment a, Microsoft.Extensions.Primitives.StringSegment b, System.StringComparison comparisonType) { throw null; }
         public bool Equals(Microsoft.Extensions.Primitives.StringSegment other, System.StringComparison comparisonType) { throw null; }
         public override bool Equals([System.Diagnostics.CodeAnalysis.NotNullWhen(true)] object? obj) { throw null; }
-        public bool Equals([System.Diagnostics.CodeAnalysis.NotNullWhen(true)] string? text) { throw null; }
-        public bool Equals([System.Diagnostics.CodeAnalysis.NotNullWhen(true)] string? text, System.StringComparison comparisonType) { throw null; }
+        public bool Equals(string? text) { throw null; }
+        public bool Equals(string? text, System.StringComparison comparisonType) { throw null; }
         public override int GetHashCode() { throw null; }
         public int IndexOf(char c) { throw null; }
         public int IndexOf(char c, int start) { throw null; }
         public int IndexOf(char c, int start, int count) { throw null; }
         public int IndexOfAny(char[] anyOf) { throw null; }
         public int IndexOfAny(char[] anyOf, int startIndex) { throw null; }
         public int IndexOfAny(char[] anyOf, int startIndex, int count) { throw null; }
+        [System.Diagnostics.CodeAnalysis.MemberNotNullWhen(false, nameof(Buffer))]
+        [System.Diagnostics.CodeAnalysis.MemberNotNullWhen(false, nameof(Value))]
```

Comment:
I don't know what the compiler will do in this case. I.e. despite saying there may be a bug here, I'm not sure it's fixable because the attributes relate to the current instance (not to the parameter's members) and `static` methods don't have an instance.

Reply:
Good catch. This doesn't work, saw your `IsNullOrEmpty` comment in the morning, quickly fixed it, and didn't noticed that it is static (checked for that in other methods though). Reverted the change.

![image](https://user-images.githubusercontent.com/6609929/131228222-3e2cc1f6-4af1-41e7-863b-43144c19fdb4.png)